changelog-extractor.py

```diff
@@ -0,0 +1,144 @@
+"""Script parses changelog from git based on 2 latest tags, formates changelog to .md format, creates/updates release on gitlab, posts to slack
+"""
+import subprocess
+import os
+
+def getMdFormattedChangelog(newCommit, oldCommit) : 
+    #saving parameters this way to make sure that they are read properly on bash command
+    get_log_format = '%x1f'.join(['%b', '%h']) + '%x1e'
+    bashCommand = 'git --no-pager --git-dir=.git log --first-parent --format="%s"' % get_log_format   + " "+ newCommit+"..." + oldCommit
+    #getting raw changelog from git
+    rawChangelog = sendCommand(bashCommand).replace("\"", "")
+    return  ConvertChangelogTextToMd(rawChangelog, None)
+
+def buildCommandForTagNotes(reqType, cleanChangeLog, tag) : 
+    formattedProjectPath = os.environ['CI_PROJECT_PATH'].replace("/","%2F")  
+    return 'curl  -X '+reqType+'  --header "PRIVATE-TOKEN: $GITLAB_API_PRIVATE_TOKEN" -d description="'+cleanChangeLog+ '" https://git.smarpsocial.com/api/v4/projects/"'+ formattedProjectPath+ '"/repository/tags/"'+tag+'"/release'
+     
+def putTagNotesOnGitlab( cleanChangeLog  , tag):
+    # create a release notes just in case it hasnt'been created before
+    k = os.system(buildCommandForTagNotes("POST", cleanChangeLog, tag))
+    # updating just in case it had been created before
+    k = os.system(buildCommandForTagNotes("PUT", cleanChangeLog, tag))
+    print "\n"
+
+def sendCommand( bashCommand ):
+    """Sends command to system.
+    """
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+    if error is not None :
+        print(error)
+        print("Terminating...")
+        quit()
+    return output.strip('\n')
+
+def getCommitByTag( tag ):
+    return  sendCommand("git rev-list -n 1 " +  tag)
+
+def parseRawChangelog(nonFormattedText ) :
+    """Parses raw changelog extracted from git.
+     Returns map {'issue_type': ['issues_array']}
+    """
+    mappedIssues = {}
+    for line in nonFormattedText.splitlines() :
+        #skipping empty lines, non related to issues description lines, etc...
+        if line == "" :
+            continue
+        if line.startswith("See merge request") or line.startswith("This reverts commit"):
+            continue
+        if len(line)<=11 and not " " in line:
+            continue
+        categorizedIssue = False
+        for issueType in issueTypes:
+            issuePrefix = issueType + ": "
+            #checking lower cased strings to prevent skipping misnamed issues
+            if line.lower().startswith(issuePrefix.lower()) :
```

Comment:
Could be nice to call strip() on the line so we are sure we ignore possible white spaces at the start

Reply:
done